state/application.go

```diff
@@ -1493,11 +1486,8 @@ func incCharmModifiedVersionOps(applicationID string) []txn.Op {
 
 func (a *Application) resolveResourceOps(resourceIDs map[string]string) ([]txn.Op, error) {
 	// Collect pending resource resolution operations.
-	resources, err := a.st.Resources()
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	return resources.NewResolvePendingResourcesOps(a.doc.Name, resourceIDs)
+	resources := a.st.Resources().(*resourcePersistence)
```

Comment:
actually maybe just add an unexported resources method that returns *resourcePersistence

Reply:
ah good idea